### PR TITLE
Classifying Alzheimers with Vision Transformer (47436761) 

### PR DESCRIPTION
**Pull request by Uncle-Mitch**
# Classifying Alzheimers with Vision Transformer (ViT)

## Overview 
This project is an implementation of a Vision Transformer (ViT) for image classification, specifically targeting the classification of brain MRI images from the ADNI dataset. The implementation covers the entire workflow from data loading to model training and predictions. You can use the code to make predictions for new images.

## Files Changed
This project is organised into several files:
- `README.md`: Provides an overview of the project, setup instructions, and usage guidelines.
- `dataset.py`: Contains the functionality for loading and preprocessing images from the ADNI dataset from a specified file path.
- `modules.py`: Implements the ViT model architecture, including patch generation, patch encoding, and the main classifier.
- `train.py`: Defines the training loop for training the ViT model on the dataset, tracking progress, and saving the trained model.

## Performance.
The model was ran for 500 epochs and achieved around 70% accuracy on the test set. The ViT model proved effective for classifying Alzheimer's Disease (AD) or Normal Control (NC) subjects based on MRI scans. Its accuracy could be improved with further training and/or more training data.

### CODE REVIEW

**Comment by shaivikaaaa** [comment]
**This is an initial inspection**

Difficulty : Hard
Task : 5

1) Recognition Problem:

- The code structure looks ok
- no images are there for brain with AD and normal 
- no graph plots are there
- the test accuracy achieved is around 0.7, minimum required is of 0.8

2) Commit Log:

- commit messages are good
- commits are progressive 


3) Documentation:

Requirement:

Tittle: Done
Description of the algorithm: Done
Problem that it solves: Done
How it works in a paragraph (example any pre-processing): Done
Figure/visualisation (any input, output & results): Done
Dependencies: Done

4) Pull Request

Can be better, please refer to PR's done in 2023 repository - for reference 

Feedback:
- update the PR

**Comment by Uncle-Mitch** [comment]
Hi @shaivikaaaa ,
Thanks for your feedback.

I will update the PR soon according to the feedback soon.

May I add the graphs / diagrams back into the readme? It seems like the assets folder was not added to the PR correctly.

**Comment by shaivikaaaa** [comment]
hi
u just need to update the current one
I believe you have already pushed 1 commit
yes, please do add graphs and diagrams in readme 

if you still push the required changes asked in feedback, you would not be marked low based on late penalty
so feel free to re-upload the images if you have it

**Comment by Uncle-Mitch** [comment]
Thanks for the reply.
I have updated the current PR and have updated the README with appropriate images.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request  includes clear description about the file structure .
Previous feedback incorporated .

File Organizing: Well-organized files.

Commit Log:
Commit messages are  progressive for the Recognition Problem solved using 4 files .
Commits show a logical developement .

Documentation:
Code comments and docstrings are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Should  add about Conclusion and future improvements in the ReadMe.

**Comment by Uncle-Mitch** [comment]
Hi, 
Thank you for marking, I appreciate the time its taken for you to look through every submission!

I don't agree with the comment for "Readme :Acceptable. No graph plots.". Graph plots were retroactively included in commit [d75fab2](https://github.com/shakes76/PatternAnalysis-2024/pull/141/commits/d75fab2ff24a7c11f0b74b610bf70fde4de18e69) after feedback from the tutor.

This fixed the broken image links in the README on github after tutor feedback. The images were present in the PDF submission, so I believe it should be valid.

**Comment by hanemma7moud** [comment]
Damon Fok

**Comment by Uncle-Mitch** [comment]
Apologies for that.
I've removed the gitignore from the root directory. 

Please let me know if anything else needs to change.
